extensions-contrib/tdigestsketch/src/test/java/org/apache/druid/query/aggregation/tdigestsketch/sql/TDigestSketchSqlAggregatorTest.java

```diff
@@ -444,7 +443,11 @@ public void testEmptyTimeseriesResults()
             Druids.newTimeseriesQueryBuilder()
                   .dataSource(CalciteTests.DATASOURCE1)
                   .intervals(new MultipleIntervalSegmentSpec(ImmutableList.of(Filtration.eternity())))
-                  .filters(bound("dim2", "0", "0", false, false, null, StringComparators.NUMERIC))
+                  .filters(
+                      NullHandling.replaceWithDefault()
+                      ? numericSelector("dim2", "0", null)
+                      : equality("dim2", 0L, ColumnType.LONG)
+                  )
```

Comment:
`numericSelector` is actually a bound filter, but there are _also_ selector filters that have number arguments, so I couldn't figure out how to consistently make the static method do the right thing. Whether or not a filter ends up as a numericSelector or regular selector in the classic mode depends on whether or not the column type is numeric (if numeric, it becomes a regular selector, if the column is a string, it stays as a bound filter with both values equal).

Reply:
You could have 2 different static methods, one for when we expect it to be running against a numerical column and one for when we do not.

That said, I'm not making this comment because I care deeply that it be done differently.  I'm just saying it because it would work.  What you have makes sense and you can also keep it like it is now.